packages/aws-cdk/lib/api/hotswap/cloudformation-executable-template.ts

```diff
@@ -0,0 +1,230 @@
+    }
+  }
+
+  async 'Fn::Sub'(template: string, explicitPlaceholders?: { [variable: string]: string }): Promise<string> {
+    const placeholders = explicitPlaceholders
+      ? { ...this.context, ...(await this.evaluateCfnExpression(explicitPlaceholders)) }
+      : this.context;
+
+    return template.replace(/\${([^}]*)}/g, (_: string, key: string) => {
+      if (key in placeholders) {
+        return placeholders[key];
+      } else {
+        throw new Error(`Fn::Sub target '${key}' was not found`);
+      }
+    });
+  }
+
+  private parseIntrinsic(x: any): Intrinsic | undefined {
+    const keys = Object.keys(x);
+    if (keys.length === 1 && (keys[0].startsWith('Fn::') || keys[0] === 'Ref')) {
+      return {
+        name: keys[0],
+        args: x[keys[0]],
+      };
+    }
+    return undefined;
+  }
+
+  private evaluateIntrinsic(intrinsic: Intrinsic): any {
+    const intrinsicFunc = (this as any)[intrinsic.name];
+    if (!intrinsicFunc) {
+      throw new Error(`CloudFormation function ${intrinsic.name} is not supported`);
+    }
+
+    const argsAsArray = Array.isArray(intrinsic.args) ? intrinsic.args : [intrinsic.args];
+
+    return intrinsicFunc.apply(this, argsAsArray);
+  }
+
+  private async findRefTarget(logicalId: string): Promise<string | undefined> {
+    // first, check to see if the Ref is a Parameter who's value we have
+    const parameterTarget = this.context[logicalId];
+    if (parameterTarget) {
+      return parameterTarget;
+    }
+    // if it's not a Parameter, we need to search in the current Stack resources
+    return this.findGetAttTarget(logicalId);
+  }
+
+  private async findGetAttTarget(logicalId: string, attribute?: string): Promise<string | undefined> {
+    const stackResources = await this.stackResources.listStackResources();
+    const foundResource = stackResources.find(sr => sr.LogicalResourceId === logicalId);
+    if (!foundResource) {
+      return undefined;
+    }
+    // now, we need to format the appropriate identifier depending on the resource type,
+    // and the requested attribute name
+    return this.formatResourceAttribute(foundResource, attribute);
+  }
+
+  private formatResourceAttribute(resource: AWS.CloudFormation.StackResourceSummary, attribute: string | undefined): string | undefined {
+    const physicalId = resource.PhysicalResourceId;
+
+    // no attribute means Ref expression, for which we use the physical ID directly
+    if (!attribute) {
+      return physicalId;
+    }
+
+    const resourceTypeFormats = RESOURCE_TYPE_ATTRIBUTES_FORMATS[resource.ResourceType];
+    if (!resourceTypeFormats) {
+      return physicalId;
+    }
+    const attributeFmtFunc = resourceTypeFormats[attribute];
+    if (!attributeFmtFunc) {
+      return physicalId;
+    }
+    const service = this.getServiceOfResource(resource);
+    const resourceTypeArnPart = this.getResourceTypeArnPartOfResource(resource);
+    return attributeFmtFunc({
```

Comment:
Oh my god the incredible hax!

Reply:
Hmm, really? This is like the mildest problem I see personally with this PR 😃.

What is your concern here?

Reply:
I misunderstood a little what was happening here. It's... okay :)